### PR TITLE
Draft Submissions Changes

### PR DESCRIPTION
**Pull request by MichaelChung123**
# Description
Made a few QOL changes to the Draft Submissions.

## Types of changes
Making the following Draft Submission Fixes:
- Added Warnings when leaving the Draft Submissions Pages
- Fixed header action button placement to be consistent with other pages with Header Action Buttons
- Added the Draft Last Edited column to the Draft Submissions Table
- Removed the Completed Date column from the Draft Submissions Table
- Spacing fix for the Delete Action Button for the Draft Submission Table Action Icons

JIRA Ref:
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-2130

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by loneil — CHANGES_REQUESTED** [review]
Tried this out on the PR with form 87e794d6-7a4a-43f1-b362-2158fff97dc2 and when saving an edit to the draft I don't see the "Draft Last Edited" field changing

File: app/frontend/src/components/forms/submission/MySubmissionsTable.vue [hunk 1]
```diff
@@ -172,13 +195,14 @@ export default {
...
-            completedDate: this.getStatusDate(s, 'COMPLETED'),
+            // completedDate: this.getStatusDate(s, 'COMPLETED'),
             confirmationId: s.confirmationId,
+            lastEdited: this.getStatusDate(s),
             name: s.name,
             permissions: s.permissions,
             status: this.getCurrentStatus(s),
             submissionId: s.formSubmissionId,
-            submittedDate: this.getStatusDate(s, 'SUBMITTED'),
+            submittedDate: s.draft ? null : this.getStatusDate(s, 'SUBMITTED')
```
Comment (loneil) [inline app/frontend/src/components/forms/submission/MySubmissionsTable.vue:205]: I don't think you'll need this check. There won't be a submitted status for draft submissions.

Comment (MichaelChung123) [reply app/frontend/src/components/forms/submission/MySubmissionsTable.vue:205]: right this makes sense, ill  remove this in next commit.

File: app/frontend/src/components/forms/submission/MySubmissionsTable.vue [hunk 2]
```diff
@@ -172,13 +195,14 @@ export default {
       if (this.submissionList) {
         const tableRows = this.submissionList.map((s) => {
           return {
-            completedDate: this.getStatusDate(s, 'COMPLETED'),
+            // completedDate: this.getStatusDate(s, 'COMPLETED'),
             confirmationId: s.confirmationId,
+            lastEdited: this.getStatusDate(s),
```
[the remark this answers is not in the document]
Comment (MichaelChung123) [reply app/frontend/src/components/forms/submission/MySubmissionsTable.vue:200]: moved this logic into an inline conditional and changed the getStatusDate function back to normal in next commit.

**Comment by MichaelChung123** [comment]
rebased

File: app/frontend/src/components/forms/submission/MySubmissionsTable.vue [hunk 3]
```diff
@@ -2,11 +2,12 @@
...
       <v-row class="mt-6" no-gutters>
-        <v-col class="text-center" cols="12" sm="10" offset-sm="1">
+        <!-- page title -->
+        <v-col cols="12" sm="6" order="2" order-sm="1">
           <h1>Previous Submissions</h1>
-          <h2>{{ formId ? form.name : 'All Forms' }}</h2>
         </v-col>
-        <v-col class="text-right" cols="12" sm="1">
+        <!-- buttons -->
+        <v-col class="text-right text-sm-right" cols="12" sm="6" order="1" order-sm="2">
```
Comment (jujaga) [inline app/frontend/src/components/forms/submission/MySubmissionsTable.vue:10]: Do we need `text-sm-right`? IIRC `text-right` implicitly means `text-xs-right` which would also include `text-sm-right`. Can't remember if that's the case or not.

**Review by jujaga — APPROVED** [review]
👍

File: app/frontend/src/components/forms/submission/MySubmissionsTable.vue [hunk 4]
```diff
@@ -172,13 +191,14 @@ export default {
       if (this.submissionList) {
         const tableRows = this.submissionList.map((s) => {
           return {
-            completedDate: this.getStatusDate(s, 'COMPLETED'),
+            // completedDate: this.getStatusDate(s, 'COMPLETED'),
             confirmationId: s.confirmationId,
+            lastEdited: s.draft ? s.createdAt : undefined,
```
Comment (TimCsaky) [inline app/frontend/src/components/forms/submission/MySubmissionsTable.vue:196]: should this be `lastEdited: s.draft ? s.updatedAt: undefined,` ?

Comment (loneil) [reply app/frontend/src/components/forms/submission/MySubmissionsTable.vue:196]: Yeah suspect this is why it's not working

Comment (MichaelChung123) [reply app/frontend/src/components/forms/submission/MySubmissionsTable.vue:196]: good catch. as discussed together, using s.createdAt does not allow you to see the updated date when you save your draft after the first time.

Will be taking a look at changing the API response to get a better field for this situation.

File: app/frontend/src/components/forms/submission/MySubmissionsTable.vue (hunk 3 above)
Comment (MichaelChung123) [reply app/frontend/src/components/forms/submission/MySubmissionsTable.vue:10]: Correct, will be changing this in the next commit. Also going to be doing this change for all the other places we are using this.

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-210

**Review by TimCsaky — APPROVED** [review]
good job.. another one down!
i guess you'll need to rebase again before merging

**Comment by MichaelChung123** [comment]
Rebased

File: app/frontend/src/components/forms/submission/MySubmissionsTable.vue [hunk 5]
```diff
@@ -119,11 +126,20 @@ export default {
...
-        headers.splice(3, 0, {
-          text: 'Completed Date',
+      // Uncomment if you want to add the Completed Date column back to the list
+      // if (this.showStatus || !this.formId) {
+      //   headers.splice(3, 0, {
+      //     text: 'Completed Date',
+      //     align: 'start',
+      //     value: 'completedDate',
+      //   });
+      // }
```
Comment (loneil) [inline app/frontend/src/components/forms/submission/MySubmissionsTable.vue:136]: Can remove comment if we're taking completed date out of requirements

Comment (MichaelChung123) [reply app/frontend/src/components/forms/submission/MySubmissionsTable.vue]: removed the unnecessary comments in new changes

**Review by loneil — APPROVED** [review]

**Pull request merged**